GRPC MockedGreeterServiceTests sample tests do not work out-of-the-box

[Edit by guardrex to add the metadata.]

# Issue description

The [`SayHelloUnaryTest_MockGreeter_Error()`](https://github.com/dotnet/AspNetCore.Docs/blob/61d48b225d0611d176d6ad4367ea50ae841b30f6/aspnetcore/grpc/test-services/sample/Tests/Server/IntegrationTests/MockedGreeterServiceTests.cs#L69-L86) integration test does not build out-of-the-box.

## How to reproduce

1. Open the repository in a GitHub CodeSpace or Visual Studio Code.

1. Move to the GRPC sample tests directory

    ```plaintext
    cd aspnetcore/grpc/test-services/sample/Tests
    ```

1. Attempt to run the tests

    ```plaintext
    dotnet test
    ```

1. You should receive a build error on `MockedGreeterServiceTests.cs`

    ```plaintext
    /workspaces/AspNetCore.Docs/aspnetcore/grpc/test-services/sample/Tests/Server/IntegrationTests/MockedGreeterServiceTests.cs(69,10): error CS0616: 'Test' is not an attribute class [/workspaces/AspNetCore.Docs/aspnetcore/grpc/test-services/sample/Tests/Tests.csproj]
    ```

## The Error

It smells like the [`Assert.Fail()`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.testtools.unittesting.assert.fail?view=visualstudiosdk-2022#microsoft-visualstudio-testtools-unittesting-assert-fail) and [`StringAssert.Contains()`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.testtools.unittesting.stringassert.contains?view=visualstudiosdk-2022) assertions that come from the `Microsoft.VisualStudio.TestTools.UnitTesting` namespace that comes packaged with Visual Studio is not referenced in [Tests.csproj](https://github.com/dotnet/AspNetCore.Docs/blob/61d48b225d0611d176d6ad4367ea50ae841b30f6/aspnetcore/grpc/test-services/sample/Tests/Tests.csproj). 

https://github.com/dotnet/AspNetCore.Docs/blob/61d48b225d0611d176d6ad4367ea50ae841b30f6/aspnetcore/grpc/test-services/sample/Tests/Server/IntegrationTests/MockedGreeterServiceTests.cs#L69-L86

## Potential Fix

We could probably instead use xUnit's exception assertions given that the rest of the tests in this sample use xUnit's `[Fact]`.

# Software versions

Check the .NET target framework(s) being used, and include the version number(s).

* [x] .NET SDK 6.0.400

<details>
<summary><strong>dotnet --info output</strong></summary>

```console
.NET SDK (reflecting any global.json):
 Version:   6.0.400
 Commit:    7771abd614

Runtime Environment:
 OS Name:     debian
 OS Version:  11
 OS Platform: Linux
 RID:         debian.11-x64
 Base Path:   /usr/share/dotnet/sdk/6.0.400/

global.json file:
  Not found

Host:
  Version:      6.0.8
  Architecture: x64
  Commit:       55fb7ef977

.NET SDKs installed:
  6.0.400 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.8 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.8 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
```
</details>

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 24ad662e-1855-15bb-8e98-698e8c29b7fb
* Version Independent ID: 24ad662e-1855-15bb-8e98-698e8c29b7fb
* Content: [Test gRPC services in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/grpc/test-services?view=aspnetcore-6.0)
* Content Source: [aspnetcore/grpc/test-services.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/grpc/test-services.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-grpc**
* GitHub Login: @JamesNK
* Microsoft Alias: **jamesnk**

--- a/aspnetcore/grpc/test-services/sample/Tests/Server/IntegrationTests/MockedGreeterServiceTests.cs
+++ b/aspnetcore/grpc/test-services/sample/Tests/Server/IntegrationTests/MockedGreeterServiceTests.cs
@@ -66,23 +66,19 @@ namespace Tests.Server.IntegrationTests
         }
         #endregion
 
-        [Test]
+        [Fact]
         public async Task SayHelloUnaryTest_MockGreeter_Error()
         {
             // Arrange
             var client = new Tester.TesterClient(Channel);
 
             // Act
-            try
-            {
-                await client.SayHelloUnaryAsync(new HelloRequest { Name = "" });
-                Assert.Fail();
-            }
-            catch (RpcException ex)
-            {
-                // Assert
-                StringAssert.Contains("Name not provided.", ex.Status.Detail);
-            }
+            var exception = await Record.ExceptionAsync(async () =>
+                await client.SayHelloUnaryAsync(new HelloRequest { Name = "" }));
+
+            // Assert
+            Assert.IsType<RpcException>(exception);
+            Assert.Contains("Name not provided.", exception.Message);
         }
     }
 }

--- a/aspnetcore/grpc/test-services/sample/Tests/Server/IntegrationTests/MockedGreeterServiceTests.cs
+++ b/aspnetcore/grpc/test-services/sample/Tests/Server/IntegrationTests/MockedGreeterServiceTests.cs
@@ -73,11 +73,10 @@ namespace Tests.Server.IntegrationTests
             var client = new Tester.TesterClient(Channel);
 
             // Act
-            var exception = await Record.ExceptionAsync(async () =>
+            var exception = await Assert.ThrowsAsync<RpcException>(async () =>
                 await client.SayHelloUnaryAsync(new HelloRequest { Name = "" }));
 
             // Assert
-            Assert.IsType<RpcException>(exception);
             Assert.Contains("Name not provided.", exception.Message);
         }
     }

